src/Traits/AuthorizationHeadersAwareTrait.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\CdtClient\Traits;
+
+use OpenEuropa\CdtClient\Model\Response\Token;
+
+/**
+ * Trait AuthorizationHeadersAwareTrait
+ *
+ * Provides methods for handling authorization headers.
+ */
+trait AuthorizationHeadersAwareTrait
+{
+    /**
+     * @return array<string, string>
+     */
+    public function getAuthorizationHeaders(?Token $token): array
+    {
+        if (null === $token) {
```

Comment:
This `if` is required by PHPStan. I had a lot of trouble re-implementing this method. In the end, I had to allow nullable tokens in the `Download` class and create additional tests for this exception.